[MIM-2304] Update postMarkdown service

olavlan-ssb opened the pull request:
**Problem:**
* Current service can only receive requests of type _/postMarkdown?displayName=my article&markdown=My markdown content_, i.e. the data must be sent as query parameters.
* However, a POST service should receive the data in the request body, according to [POST (HTTP)](https://en.wikipedia.org/wiki/POST_(HTTP)) .

**Solution:**
* The service now accepts a request containing a JSON-formatted string in the body.
* The service accepts keys *_id*, *markdown* and *displayName* in the JSON string. 

**Limitations:**
* The exception where the body can't be parsed as JSON is not explicitly handled. Should it be handled, and if yes, how? @johnnadeluy 

### Jira Issues!
[Link to ticket: MIM-2304](https://statistics-norway.atlassian.net/browse/MIM-2304)

### Branch Deployment to NAIS!
[🗄️ Link to branch admin interface!](https://ssbno-mim-2304.intern.test.ssb.no/admin)
[📰 Front page can be found here!](https://ssbno-mim-2304.intern.test.ssb.no/)

annesiri commented:
Vi har problem med test-miljøet så det er derfor testene feiler. Så det er sannsynsligvis ingenting med din kode som trigger feilene der.

johnnadeluy reviewed (changes requested):
To answer the question; yes, we should have conditionals/checks when `req.body` is null or returns an unexpected type after parse (e.g. const json returns a string or array instead of an object, which will cause `json._id` in `data` to cause an error). For example:

``` const json = req?.body ? JSON.parse(req.body) : {} ```

So we should to ensure that `const json` is an object, and that we have fallbacks for fields `_id`, `markdown`, and `displayName` when `json` returns an empty object ({}). It's similar to what we do for `nodeId` and `nodeExists` (Line 14 & 15). For example:

```const data = {
_id: json?._id ?? '',
markdown: json?.markdown ?? '',
displayName: json?.displayName ?? ''
}```

olavlan-ssb commented:
> > The exception where the body can't be parsed as JSON is not explicitly handled. Should it be handled, and if yes, how?
> 
> Yes, we should have conditionals/checks when `req.body` is null or returns an unexpected type after parse (e.g. const json returns a string or array instead of an object, which will cause `json._id` in `data` to cause an error). For example:
> 
> `const json = req?.body ? JSON.parse(req.body) : {}`
> 
Fixed in 4adae25. 

After doing this I realised that `json._id` is only accessed once and it has no purpose to store it in the `data` object (in fact, it makes no sense to pass it when creating or modifying a node). I fixed this in 0d74c64.

> So we should to ensure that `const json` is an object, and that we have fallbacks for fields `_id`, `markdown`, and `displayName` when `json` returns an empty object ({}). It's similar to what we do for `nodeId` and `nodeExists` (Line 14 & 15). For example:
> 
> ```
> const data = {
>     _id: json?._id ?? '',
>     markdown: json?.markdown ?? '',
>     displayName: json?.displayName ?? ''
> }
> ```

Fixed in a0c5d1c. Since we ensured that `json` is an object (i.e. never `null`), is it enough to access its attributes like `json.markdown ?? ''` ?

johnnadeluy commented:
> Since we ensured that `json` is an object (i.e. never `null`), is it enough to access its attributes like json.markdown ?? '' ?

I think that this fallback should be enough so we don't get `undefined` and `null` errors when the properties are references for rendering/listing. But now that you mention it, we should have better fallback text so that the response data isn't invisible? 🤔

johnnadeluy reviewed (approved):
Yes, that sounds like a good idea! ☺️ Approving this PR ✅ Good job!
